### PR TITLE
Browser v2

### PR DESCRIPTION
**Pull request by dakota002**
Removing local dependency on schema repo. Still need to add a means of selecting versions

### CODE REVIEW

**Comment by dakota002** [comment]
This will require a github api token for the site before merging

File: app/routes/docs/contributing/index.mdx [hunk 1]
```diff
@@ -97,6 +97,11 @@ Here are instructions for getting the GCN site set up and running on your own co
     # RECAPTCHA_SITE_KEY=fill-me-in
     # RECAPTCHA_SECRET_KEY=fill-me-in
 
+    # (Optional) Github Integrations
+    # For parts of the site that pull data using Github's content api (i.e. schema-browser)
+    # See https://github.com/settings/tokens to generate an api token
```
Comment (lpsinger) [inline app/routes/docs/contributing/index.mdx:102]:
```suggestion
    # For parts of the site that pull data using Github's content API (i.e. schema-browser)
    # See https://github.com/settings/tokens to generate an API token
```

File: app/lib/schema-data.ts [hunk 2]
```diff
@@ -52,19 +60,34 @@ export async function loadJson(filePath: string): Promise<Schema> {
...
+  const ghData = (
+    await octokit.repos.getContent({
+      ...repoData,
+      path: path.replaceAll('\\', '/'),
+      ref: ref ?? 'main',
+    })
+  ).data as GitContentDataResponse
+  if (!ghData.content) throw new Response(null, { status: 404 })
+
+  return JSON.parse(Buffer.from(ghData.content, 'base64').toString()) as Schema
```
Comment (lpsinger) [inline app/lib/schema-data.ts:73]: I'm surprised that it returns the content base64-encoded. Can you avoid the decoding step by setting the [media type](https://github.com/octokit/octokit.js#media-type-formats)?

File: app/routes/docs/schema-browser/$.tsx [hunk 3]
```diff
@@ -22,17 +24,24 @@ import {
 } from '~/routes/docs/schema-browser/SchemaBrowserElements.lib'
 
 export async function loader({ params: { '*': path } }: DataFunctionArgs) {
-  if (!path) throw new Response(null, { status: 404 })
+  if (!path || !path.includes('.schema.json'))
```
Comment (lpsinger) [inline app/routes/docs/schema-browser/$.tsx:27]:
```suggestion
  if (!path?.includes('.schema.json'))
```

File: app/lib/schema-data.ts [hunk 4]
```diff
@@ -73,26 +96,59 @@ export type ExampleFiles = {
...
+      if (item.type == 'dir' || item.type == 'submodule') {
+        item.children = await getContentTree(item.path, ref)
+      }
+    }
+
+    return data
+  } catch (error) {
+    return []
+  }
+}
```
Comment (lpsinger) [inline app/lib/schema-data.ts:154]: I'm troubled by the performance implications of recursively walking the entire directory tree. It occurs to me that if the schema browser side nav supports expanding _only a single directory at a time_, then we only need to return the listing of a single directory in any one loader invocation.

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please address the outstanding comments from the last review. Sorry, this is a big PR, it's going to take me a few iterations to spot everything.

**Comment by dakota002** [comment]
Still working on the changes from the feedback. The reworking of the getContentTree to not be recursive is just taking a bit longer than I'd like, but should have it done soon

**Comment by dakota002** [comment]
Closed by #1130

**Pull request closed without merging**